src/data/nat/basic.lean

```diff
@@ -1171,12 +1171,10 @@ lemma dvd_of_pow_dvd {p k m : ℕ} (hk : 1 ≤ k) (hpk : p^k ∣ m) : p ∣ m :=
 by rw ←nat.pow_one p; exact pow_dvd_of_le_of_pow_dvd hk hpk
 
 lemma eq_of_dvd_quot_one {a b : ℕ} (w : a ∣ b) (h : b / a = 1) : a = b :=
-begin
-  rcases w with ⟨b, rfl⟩,
-  rw [nat.mul_comm, nat.mul_div_cancel] at h,
-  { simp [h] },
-  { by_contradiction, simp * at * }
-end
+by rw [←nat.div_mul_cancel w, h, one_mul]
+
+lemma eq_of_dvd_quot_zero {a b : ℕ} (w : a ∣ b)  (h : b / a = 0) : b = 0 :=
```

Comment:
We don't use `quot` for natural number division, we use `div`. I suggest `eq_zero_of_dvd_of_div_eq_zero`

Reply:
Done!